### PR TITLE
Week 6 surielis rodriguez

### PR DESCRIPTION
**Pull request by SurielisRodriguez**
# Submitting Week 6 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ x] Coding Challenges
- [ x] Coding Concepts
- [ x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,6 +14,12 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
+// describe('sentence', () => {
+//   it("capitalizes each name", () => {
+//     expect(sentence).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
```
Comment (chrisyhlee) [inline code-challenges.test.js:19]: You should be testing the output of your function, not a variable.

File: code-challenges.test.js [hunk 2]
```diff
@@ -23,6 +29,31 @@ const hitchhikersCharacters = [
...
+// declare var is the key we want to split within our  character object, we split by the space
+// map the new array and pass in parameter for word
+// return and tell our program to upper case the value at zero index using the .charAt() and .toUpperCase () methods
+// add back the rest of the name starting at the 1 index using .slice(1) to add back the rest of the name
+// declare a second variable for our full name put together and pass the last split arrays we made through to put them together using the .join method
+// using string interpolation we tell the program to return the value for the fullName that we created and the characters occupation to put or sentence together
+// console log to see expected outcome.
+
+
+const sentence = hitchhikersCharacters.map(character => {
```
Comment (chrisyhlee) [inline code-challenges.test.js:43]: This is not a function, but it is a variable!

File: code-challenges.test.js [hunk 3]
```diff
@@ -34,13 +65,56 @@ const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
...
+// map the filtered array to be able to assign a second variable which I designated as my remain,  then return that var is the remainder of the number divided by 3 
+// return filter val
+//console log to see expected outputs. 
+
+const remainders = (array) => {
+  let filter = array.filter(item => typeof item === 'number' ).map(number => {
+    let remain = number % 3
+    return remain
+  })
+  return filter
```
Comment (chrisyhlee) [inline code-challenges.test.js:80]: Definitely some refactors we can make here to get rid of all the variables

File: code-challenges.test.js [hunk 4]
```diff
@@ -34,13 +65,56 @@ const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
...
+// create method called cuber that takes the array as a parameter.
+// create variable in which numbers array gets mapped and with the variable of num that returns the values as numbers instead of string by using the parseInt method.
+// return cube number by using exponential operator
+// return the numbers and using the .reduce method we can take all of the numbers within my array and makes it into one element, specifying that I wanted each number to add to the next. 
+// console log to see expected output.
+
+
+const cuber = (numArray) => {
+  let numbers = numArray.map( num => {
+    return parseInt(num) ** 3
```
Comment (chrisyhlee) [inline code-challenges.test.js:115]: The original array contains numbers already, not strings, so parseInt is not necessary here

File: interview-questions.md [hunk 5]
```diff
@@ -8,36 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: The foreign key can be added by adding a column called cohort_id:integer so that every student has a cohort and is able to be identified and referenced by that ID. Setting up the foreign Id helps link these two models to each other.
 
-Researched answer:
+Researched answer: The foreign ID is a attribute passed that refers to the primary key of a different model. It will link both models and you will be able to access information on both tables. Tables only allow one primary key so this a tool used to build a relationship between both tables.
```
Comment (chrisyhlee) [inline interview-questions.md:13]: I would include that in order to add a column for the foreign key, you need to generate a migration.

File: rails-commenting.rb [hunk 6]
```diff
@@ -6,39 +6,39 @@
...
   end
 
-  # ---3)
+  # ---3)This is creating a method for show which will bring up a specific instance in our databse and we define show as a post that is refrenced by the params ID
   def show
     @post = BlogPost.find(params[:id])
   end
 
-  # ---4)
+  # ---4)This is creating a method for new which creates a new instance in our application of a new post
```
Comment (chrisyhlee) [inline rails-commenting.rb:21]: This will display a form for the user to fill out to create a new blog post

File: rails-commenting.rb [hunk 7]
```diff
@@ -48,15 +48,15 @@ def update
...
     @post = BlogPost.find(params[:id])
     if @post.destroy
-      # ---8)
+      # ---8)This is saying if our blog post is deleted, then its going to redirect us to blog posts path we've created.
       redirect_to blog_posts_path
     end
   end
 
-  # ---9)
+  # ---9)This is where we designate our strong parameters, which dictate what information can be passed into the database and checks for validity of each entry and its corresponding values that we haev set in our model.
```
Comment (chrisyhlee) [inline rails-commenting.rb:56]: The private keyword makes anything below it only available in the scope of this controller

File: interview-questions.md [hunk 8]
```diff
@@ -8,36 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: The foreign ID is a attribute passed that refers to the primary key of a different model. It will link both models and you will be able to access information on both tables. Tables only allow one primary key so this a tool used to build a relationship between both tables. 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Create, Show, Edit, Update, and Destroy. These are used to create HTTP requests to the server. 
 
-Researched answer:
+Researched answer: Parameters play an important part in communicating essential information the server needs to let it perform the specified task accurately. They provide extra information to the server so they can be used in a multitude of ways and can also modify the behavior of the API endpoint.
```
Comment (chrisyhlee) [inline interview-questions.md:19]: The routes would be show, edit, update, destroy. These RESTful routes are targeting one specific primary key and performing their action on only the targeted id.

**Review by chrisyhlee — APPROVED** [review]
You have completed your final assessment, yay!! Very nice work getting to this point! It has been a journey seeing you become a stronger developer each week, and this week is no exception. I definitely see your logic flowing more smoothly, and your pseudo code is getting better at explaining each step of the process. I left some feedback to help you throughout the assessment, so be sure to take a look. Overall this displays Novice level, keep working hard on these upcoming projects you will be building!

**Comment by SurielisRodriguez** [comment]
Thanks for the Feedback Chris!

**Pull request merged**